### PR TITLE
Create Operator Subscription Improvements

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

An [Operator's provided APIs are now visible on the `PackageManifest`](https://github.com/operator-framework/operator-lifecycle-manager/pull/754) object, which can be used for a richer experience by preventing Operator conflicts and displaying which APIs will be added after installing an Operator.

Subscribing to an Operator now has a single flow through the "OperatorHub" view. Creating a `CatalogSource` in a namespace will make those packages available when viewing "OperatorHub" with that namespace active (labeled as "Custom").

### Changes

- Check [`olm.providedAPIs` annotation on `OperatorGroup`](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/operatorgroups.md#provided-apis-annotation) does not intersect with provided APIs of selected `PackageManifest` when creating `Subscription`
- No longer create/update a `CatalogSourceConfig` when subscribing to a package in OperatorHub
- Create `Subscription` to `CatalogSource` providing the selected package directly
- OperatorHub displays all `PackageManifests` for the current namespace, not just `openshift-marketplace`

### Testing

1. Create `CatalogSource` using image with **Prometheus Operator (Copy)** package (`quay.io/alecmerdler/upstream-community-operators:MKTPLC-367`)
2. Create `Subscription` to normal **Prometheus Operator**
3. Attempt to create `Subscription` to **Prometheus Operator (Copy)** package

### Screenshots

**Create Subscription Form (conflicting provided APIs):**
Screenshot_20190523_145600

### Future Work

- `CatalogSource` management using `CatalogSourceConfigs` in the **Operator Management** tab
- Visualize Operator dependencies using `customresourcedefinitions.required`

~**Blocked by** https://github.com/operator-framework/operator-lifecycle-manager/pull/871~
~**Blocked by** https://github.com/operator-framework/operator-lifecycle-manager/pull/890~
Addresses https://jira.coreos.com/browse/MKTPLC-367
Addresses https://jira.coreos.com/browse/OLM-1090

### CODE REVIEW

File: frontend/integration-tests/tests/olm/update-channel-approval.scenario.ts [hunk 1]
```diff
@@ -1,88 +0,0 @@
-import { browser, ExpectedConditions as until, $, by, element } from 'protractor';
```
Comment (alecmerdler) [inline frontend/integration-tests/tests/olm/update-channel-approval.scenario.ts]: This scenario is outdated, so removing it.

File: frontend/integration-tests/tests/operator-hub/operator-hub.scenario.ts [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
 import * as crudView from '../../views/crud.view';
 import * as catalogView from '../../views/catalog.view';
 import * as catalogPageView from '../../views/catalog-page.view';
-import * as operatorHubView from '../../views/operator-hub.view';
-
-describe('Subscribing to an Operator from OperatorHub', () => {
-  let installedOperator: string;
+import * as sidenavView from '../../views/sidenav.view';
+
+describe('Interacting with OperatorHub', () => {
```
Comment (alecmerdler) [inline frontend/integration-tests/tests/operator-hub/operator-hub.scenario.ts]: This scenario now just tests the tile view functionality (filtering, sorting, etc). Actual Operator installation are found in the other two tests.

**Comment by spadgett** [comment]
cc @jwforres @tlwu2013

**Comment by alecmerdler** [comment]
@aravindhp This PR requires that the Marketplace Operator migrates existing `Subscriptions` to point at the single `CatalogSource` in `openshift-marketplace` instead of the "installed" one in their namespace. Otherwise, we see `PackageManifests` show up twice for a given namespace.

**Comment by aravindhp** [comment]
@alecmerdler it is on our to-do list. We are first working on removing connection between `OperatorSources` and the datastore CatalogSourceConfigs. We will tackle migrating existing `Subscriptions` after that. It is being tracked by [MKTPLC-398](https://jira.coreos.com/browse/MKTPLC-398)

**Comment by spadgett** [comment]
@nicolethoen fyi

File: frontend/integration-tests/tests/olm/global-installmode.scenario.ts [hunk 3]
```diff
@@ -15,52 +17,93 @@ describe('Interacting with the Redis Operator (all-namespaces install mode)', ()
   const redisOperatorName = 'redis-enterprise-operator';
   const testLabel = 'automatedTestName';
   const redisEnterpriseCluster = `${testName}-redisenterprisecluster`;
-  const operatorGroupName = 'test-global-operatorgroup';
+
+  // Only way to find the "global" OLM catalog namespace is from container args
```
[the remark this answers is not in the document]
Comment (alecmerdler) [reply frontend/integration-tests/tests/olm/global-installmode.scenario.ts]: Alright, changed to use `--parameters.globalCatalogNamespace` and `--parameters.globalOperatorsNamespace` with OKD defaults set.

**Comment by spadgett** [comment]
@alecmerdler let me know when this is ready for another review

**Pull request merged**